tools/xcodeproj_shims/installers/swiftmodules.sh

```diff
@@ -4,8 +4,21 @@ set -euo pipefail
 
 # Copy Bazel build `.swiftmodule` files to `DerivedData`. This is used by Xcode
 # and its indexing.
-find "$BAZEL_WORKSPACE_ROOT/bazel-out"/*/bin/ \
-    -name "*.swiftmodule" \
-     -not -path "*/_swift_module_cache/*" \
-     -print0 \
-    | xargs -0 "$BAZEL_INSTALLERS_DIR/_swiftmodule.sh"
+echo "Start copying swiftmodules at `date`"
+FOUND_SWIFTMODULES=`pcregrep -o1 'primary_output: "(.*\.swiftmodule)' $BAZEL_BUILD_EVENT_TEXT_FILENAME | uniq`
```

Comment:
I wonder what's the difference between `primary_output` and `important_output`....

Reply:
After some digging, I learned that they are the same thing, but that `important_output` is deprecated:

https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto

```
// Temporarily, also report the important outputs directly. This is only to
  // allow existing clients help transition to the deduplicated representation;
  // new clients should not use it.
  repeated File important_output = 4 [deprecated = true];
```

```
  // Identifier of an event reporting that an action was completed (not all
  // actions are reported, only the ones that can be considered important;
  // this includes all failed actions).
  message ActionCompletedId {
    string primary_output = 1;
    // Optional, the label of the owner of the action, for reference.
    string label = 2;
    // Optional, the id of the configuration of the action owner.
    ConfigurationId configuration = 3;
  }
```